core-tests/src/test/java/org/mule/runtime/core/processor/chain/DefaultMessageProcessorChainTestCase.java

```diff
@@ -743,37 +764,25 @@ protected Event process(Processor messageProcessor, Event event) throws Exceptio
       ((MuleContextAware) messageProcessor).setMuleContext(muleContext);
     }
     if (messageProcessor instanceof FlowConstructAware) {
-      ((FlowConstructAware) messageProcessor).setFlowConstruct(mockFlow);
+      ((FlowConstructAware) messageProcessor).setFlowConstruct(flow);
     }
     try {
-      if (nonBlocking) {
-        try {
-          return just(event)
-              .transform(messageProcessor)
-              .subscribe()
-              .blockMillis(RECEIVE_TIMEOUT);
-        } catch (Throwable exception) {
-          throw (Exception) unwrap(exception);
-        }
-      } else {
-        return messageProcessor.process(event);
-      }
+      return super.process(messageProcessor, event);
     } finally {
       final SimpleUnitTestSupportSchedulerService schedulerService =
           (SimpleUnitTestSupportSchedulerService) (muleContext.getSchedulerService());
-      new PollingProber().check(new JUnitLambdaProbe(() -> {
-        assertThat(schedulerService.getScheduledTasks(), greaterThanOrEqualTo(nonBlockingProcessorsExecuted.get()));
-        return true;
-      }));
+      if (processingStrategyFactory instanceof LegacyNonBlockingProcessingStrategyFactory && isReactive()) {
```

Comment:
This is just updating existing code. Going forward this test case shouldn't be testing threading, but I didn't want to remove coverage just yet.